### PR TITLE
Update mipsy_cli to clapV4

### PR DESCRIPTION
**Pull request by Dylan-Brotherston**
Update clap to v4

And add a [work-around](https://docs.rs/gloo-utils/latest/gloo_utils/format/trait.JsValueSerdeExt.html) to a [deprecation in wasm_bindgen](https://docs.rs/wasm-bindgen/latest/wasm_bindgen/struct.JsValue.html#method.from_serde)

### CODE REVIEW

File: crates/mipsy_web/Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ crate-type = ["cdylib", "rlib"]
 yew = "0.19.3"
 yew-agent = "0.1.0"
 gloo-file = "0.2.0"
+gloo-utils = { version = "0.1.5", features = ["serde"]}
```
Comment (SpanishPear) [inline crates/mipsy_web/Cargo.toml:15]: Where/ why is this being used? 

I can see that https://github.com/insou22/mipsy/pull/251/files#diff-c6996f00d87c10aab85be0b286226529ab6ebc71d826ef80d1d13d4f78e5d6f7R8 imports it, but what for?

File: crates/mipsy/Cargo.toml [hunk 2]
```diff
@@ -11,15 +12,10 @@ mipsy_lib = { version = "0.1.0", path = "../mipsy_lib" }
 mipsy_parser = { version = "0.1.0", path = "../mipsy_parser" }
 mipsy_interactive = { version = "0.1.0", path = "../mipsy_interactive" }
 mipsy_utils = { version = "0.1.0", path = "../mipsy_utils" }
-mipsy_instructions = { path = "../mipsy_instructions", features = ["rt_yaml"] }
-clap = { version = "3.1.6", features = ["derive"] } # cli arg parsing
-rustyline = "9.1.1"         # readline support for interactive mipsy
```
Comment (SpanishPear) [inline crates/mipsy/Cargo.toml:16]: where did these dependancies go? :thinking: 

Or are they supposed to be in mipsy_interactive instead, and hence these are useless?

File: crates/mipsy_web/Cargo.toml (hunk 1 above)
Comment (Dylan-Brotherston) [reply crates/mipsy_web/Cargo.toml:15]: `JsValue::from_serde` is no longer implemented by wasm_bindgen.

File: crates/mipsy/Cargo.toml (hunk 2 above)
Comment (Dylan-Brotherston) [reply crates/mipsy/Cargo.toml:16]: These are mipsy_interactive dependencies. They aren't used by this crate.

File: crates/mipsy/src/main.rs [hunk 3]
```diff
@@ -11,39 +11,42 @@ use mipsy_utils::{MipsyConfig, MipsyConfigError, config_path, read_config};
...
     hex_pad_zero: bool,
 
     /// Enable some SPIM compatibility options
-    #[clap(long)]
+    #[arg(long)]
     spim: bool,
 
     /// Move a label to point to a different label
-    #[clap(long)]
+    #[arg(long, hide = true)]
```
[the remark this answers is not in the document]
Comment (Dylan-Brotherston) [reply crates/mipsy/src/main.rs:41]: Was testing something out and forget to remove it.

File: crates/mipsy_web/Cargo.toml (hunk 1 above)
Comment (SpanishPear) [reply crates/mipsy_web/Cargo.toml:15]: ooof course, lets depreciate an api and make a minor version bump

Comment (insou22) [reply crates/mipsy_web/Cargo.toml:15]: no such thing as a minor bump on 0.x.y semver --- minor is considered breaking by default until 1.x.y

**Review by SpanishPear — APPROVED** [review]

**Comment by SpanishPear** [comment]
@Dylan-Brotherston a few conflicts but otherwise good to merge i think

**Comment by Dylan-Brotherston** [comment]
Great, will check in a couple of minutes.

**Pull request merged**